pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -140,6 +141,8 @@ public BaseBrokerRequestHandler(PinotConfiguration config, RoutingManager routin
 
     _enableQueryLimitOverride = _config.getProperty(Broker.CONFIG_OF_ENABLE_QUERY_LIMIT_OVERRIDE, false);
 
+    _serveOfflineSegmentsImmediately = _config.getProperty(Broker.CONFIG_OF_BROKER_SERVE_OFFLINE_SEGMENTS_IMMEDIATELY, false);
```

Comment:
this should be a table level config. We don't want this to apply to all the tables in the system.

Reply:
@kishoreg, that makes sense. Any recommendation for which subconfig of table config? There are routing configs, segment configs, etc. I'm not sure the right one.

Reply:
I've done the refactor to add to the QueryConfig, but let me know if a different subconfig makes more sense. It won't be difficult to move to a different config class and it should look structurally the same.